inc/packages/namespace.php

```diff
@@ -0,0 +1,580 @@
+ *
+ * @return string
+ */
+function get_packages_dir() : string {
+	$path = WP_CONTENT_DIR . '/uploads/es-packages';
+
+	/**
+	 * Filter the path at which Elasticsearch package files are stored.
+	 *
+	 * @param string $path Absolute directory path, defaults to wp-content/es-packages.
+	 */
+	$path = apply_filters( 'altis.search.packages_dir', $path );
+
+	return $path;
+}
+
+/**
+ * Get a package file path.
+ *
+ * @param string $slug A slug ID for the package file.
+ * @param bool $for_network If true returns path for network level package.
+ * @return string
+ */
+function get_package_path( string $slug, bool $for_network = false ) : string {
+	$path = sprintf(
+		'%s/%s-%s.txt',
+		get_packages_dir(),
+		$for_network ? 'global' : 'site-' . get_current_blog_id(),
+		$slug
+	);
+
+	return $path;
+}
+
+/**
+ * Get the package ID for a given file path.
+ *
+ * @param string $slug The package slug to get the package ID for.
+ * @param bool $for_network If true get the network level package ID.
+ * @return string|null
+ */
+function get_package_id( string $slug, bool $for_network = false ) : ?string {
+	if ( $for_network ) {
+		$package_id = get_site_option( "altis_search_package_{$slug}", null );
+	} else {
+		$package_id = get_option( "altis_search_package_{$slug}", null );
+	}
+
+	/**
+	 * Filter the returned package ID for a given slug.
+	 *
+	 * @param string|null $package_id The package ID to return.
+	 * @param string $slug The package slug.
+	 * @param bool $for_network True for network level package.
+	 */
+	$package_id = apply_filters( 'altis.search.get_package_id', $package_id, $slug, $for_network );
+
+	return $package_id;
+}
+
+/**
+ * Process the packages form.
+ *
+ * @return void
+ */
+function handle_form() : void {
+	if ( ! isset( $_POST['action'] ) || $_POST['action'] !== 'altis-search-config' ) {
+		return;
+	}
+
+	if ( ! check_admin_referer( 'altis-search-config', '_altisnonce' ) ) {
+		return;
+	}
+
+	// Track errors.
+	$errors = [];
+
+	// Ensure upload directory exists.
+	$base_dir = get_packages_dir();
+	if ( ! file_exists( $base_dir ) ) {
+		$package_dir_created = wp_mkdir_p( $base_dir );
+		if ( ! $package_dir_created ) {
+			$errors[] = new WP_Error(
+				'create_package_dir_failed',
+				// translators: %s replaced by package base directory.
+				sprintf( __( 'Failed to create packages base directory at %s', 'altis' ), $base_dir )
+			);
+		}
+	}
+
+	// Set a flag for whether this is site level or network level.
+	$for_network = is_network_admin();
+
+	// Track whether the index needs to be updated after any package updates.
+	$should_update_indexes = false;
+
+	// The different types of package.
+	$types = [ 'synonyms', 'stopwords', 'user-dictionary' ];
+
+	foreach ( $types as $type ) {
+		$text_field = "{$type}-text";
+		$file_field = "{$type}-file";
+		$remove_field = "{$type}-remove";
+		$package_id = null;
+
+		// Handle manual entry.
+		if ( isset( $_POST[ $text_field ] ) && ! empty( $_POST[ $text_field ] ) ) {
```

Comment:
This is redundant, can be simplified to:

```suggestion
		if ( ! empty( $_POST[ $text_field ] ) ) {
```